Lib/idlelib/editor.py

```diff
@@ -1633,7 +1628,8 @@ def readline(self):
     def tokeneater(self, type, token, start, end, line,
                    INDENT=tokenize.INDENT,
                    NAME=tokenize.NAME,
-                   OPENERS=('class', 'def', 'for', 'if', 'try', 'while')):
+                   OPENERS=('class', 'def', 'for', 'if', 'match','try',
```

Comment:
```suggestion
                   OPENERS=('class', 'def', 'for', 'if', 'match', 'try',
```
Also, perhaps `async` should be included? Not familiar with what this code does.

Reply:
It looks for a compound statement header.  If I remember correctly that 'async' is always followed by one of the other words, it might be appropriate to include it.  I will test what tokenize does with such a header.

Reply:
Experiments help.  The class looks for the first line of a compound statement followed by the first line of an indented suite.  (Header continuation lines do not matter because they cannot legally contain keywords as names and do not generate INDENT tokens.)  OPENERS does not need 'async' as the following keyword will trigger anyway.  The difference in indents, in spaces, is used as a guess of the indent size.  Thank you for reviewing.